README.md

```diff
@@ -1,41 +1,149 @@
+ *Services* are the building blocks of your project. Services are defined in the .servicer/services.yaml file in your project, and these definitions include the configuration and commands needed to build, test and deploy each service. Additionally, service dependencies may be defined to automate the order services and steps are executed in.
 
-### Project Service Configuration ###
-Your project's `services.yaml` file represents the settings and service relationships unique to your project. This configuration resides in source control.
+The *project environment* should contain environment variables for holding credentials and other secrets that should not be committed to source control. These can be configured on your CI tool, or, if you are running Servicer locally, from a `.servicer/*.env.yaml` file.
 
-### Environment Settings ###
-Any environment specific credentials or secrets should live in simple environment variables, which can be read and inserted into the build at build-time. These secrets should never be checked into source control, and instead be populated in the build environment.
+## Using Servicer ##
+For the simplest use case, use the command below. With it, Servicer's *job* will be to take every step for every service you've defined in `.servicer/services.yaml`.
+```
+servicer
+```
+If you want Servicer to execute all steps for one or more specific services, like my-docker-image and my-python-package, specify those services in the command.
+```
+servicer --service=my_docker_image,my_python_package
+```
+Steps, like build and unit_test, can also be specified.
+```
+servicer --step=build,unit_test
+```
+These commands can be run locally or in the context of a CI job.
 
-## services.yaml ##
-The services config file supports environment variable substitution using the `${MY_VAR}` format.
+## Configuration ##
+`.servicer/services.yaml` must be present at the root of your project. `servicer/builtin/defaults.yaml` (in this repository) contains default values and descriptions of the values you can provide in `.servicer/services.yaml`.
+
+Within this *Configuration* section of the README, consider each of these subsections to compose the `services.yaml` on one project.
+
+### Environment ###
+The `environment` section of `services.yaml` will let you define environment variables that you're ok with existing in source control. Additionally, you can control the branch -> service_environment settings, allowing you to change the defaults. `services.yaml` supports environment variable substitution using the `${MY_VAR}` format. Consider the example below.
 
-Example config file for a deploying an AWS RDS instance and GCloud Kubernetes cluster from Bitbucket Pipelines CI:
 ```
-ci:
-  providers:
-    - bitbucket
-  image: python:3.6.4
+environment:
+  # In this example, SERVICE_ENVIRONMENT will be 'production' if the branch is 'master'.
+  # Otherwise, SERVICE_ENVIRONMENT will be the sanitized branch name.
```

Comment:
This comment doesn't make complete sense without a little more clarification:

```
# In this example, SERVICE_ENVIRONMENT will be `production` if the branch is `master`.
# SERVICE_ENVIRONMENT will be `develop` if the branch is `develop`.
# Otherwise, if the branch matches `env-*`, SERVICE_ENVIRONMENT will be the sanitized branch name.
```